spec/classes/foreman_spec.rb

```diff
@@ -41,5 +41,92 @@ class { 'foreman::providers':
         it { should contain_package('apipie-bindings').that_subscribes_to('Class[foreman::repo]') }
       end
     end
+
+    describe 'with all parameters' do
+      let :params do {
+          :foreman_url               => 'http://localhost',
+          :unattended                => true,
+          :authentication            => true,
+          :passenger                 => true,
+          :passenger_ruby            => '/usr/bin/ruby',
+          :passenger_ruby_package    => 'ruby-gem-passenger',
+          :use_vhost                 => true,
+          :servername                => 'localhost',
+          :ssl                       => true,
+          :custom_repo               => false,
+          :repo                      => 'nightly',
+          :configure_epel_repo       => true,
+          :configure_scl_repo        => false,
+          :selinux                   => true,
+          :gpgcheck                  => true,
+          :version                   => '1.12',
+          :db_manage                 => true,
+          :db_type                   => 'postgresql',
+          :db_adapter                => 'UNSET',
+          :db_host                   => 'UNSET',
+          :db_port                   => 'UNSET',
+          :db_database               => 'UNSET',
+          :db_username               => 'foreman',
+          :db_password               => 'secret',
+          :db_sslmode                => 'UNSET',
+          :db_pool                   => 5,
+          :apipie_task               => 'apipie:cache:index',
```

Comment:
The signature of init.pp changed, I guess that's a leftover of the old base you had that PR on initially.